### PR TITLE
Fix for Feh's right-click buffer overflow crash

### PR DESCRIPTION
**Pull request by CharlzKlug**
## Description of changes

Added a patch for fix the right-click crash. https://github.com/NixOS/nixpkgs/issues/258444

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FliegendeWurst** [comment]
Thanks for the fix.
Please change the commit message so it follows the guidelines in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md.
Also link your PR https://github.com/derf/feh/pull/723 in a comment next to the patch.

**Comment by CharlzKlug** [comment]
> Also link your PR [derf/feh#723](https://github.com/derf/feh/pull/723) in a comment next to the patch.

Thank you for your reply! Sorry, I can't understand where I should link my PR. Can you give me a clue?

File: pkgs/applications/graphics/feh/default.nix [hunk 1]
```diff
@@ -14,6 +14,8 @@ stdenv.mkDerivation rec {
     hash = "sha256-9NJ6zgQHcFJPmRlqJuCMXcKjLvDPUG+QvKGTJlWvWK4=";
   };
 
+  patches = [ ./0001-Right-click-crash-fix.patch ];
```
Comment (FliegendeWurst) [inline pkgs/applications/graphics/feh/default.nix:17]:
```suggestion
  patches = [
    # upstream PR: https://github.com/derf/feh/pull/723
    ./0001-Right-click-crash-fix.patch
  ];
```

Comment (CharlzKlug) [reply pkgs/applications/graphics/feh/default.nix]: Thank you! I have added link for the PR, and changed the commit message.

**Comment by FliegendeWurst** [comment]
Looks good now, thank you. Please squash your commits (see [here](https://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) if you don't know how).

**Comment by CharlzKlug** [comment]
> Looks good now, thank you. Please squash your commits (see [here](https://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) if you don't know how).

I have tried to squash the commits.

**Review by FliegendeWurst — APPROVED** [review]

**Review by eclairevoyant — APPROVED** [review]
Easily reproducible crash and the patch does indeed fix it. Thanks for the contribution!

**Comment by CharlzKlug** [comment]
> Easily reproducible crash and the patch does indeed fix it. Thanks for the contribution!

You are welcome!

**Comment by Artturin** [comment]
We prefer to `fetchpatch` patch when there's an upstream PR, I've switched to fetchpatch for you.

File: pkgs/applications/graphics/feh/default.nix [hunk 2]
```diff
@@ -14,6 +14,11 @@ stdenv.mkDerivation rec {
     hash = "sha256-9NJ6zgQHcFJPmRlqJuCMXcKjLvDPUG+QvKGTJlWvWK4=";
   };
 
+  patches = [
+    # upstream PR: https://github.com/derf/feh/pull/723
+    ./0001-Right-click-crash-fix.patch
```
Comment (ambroisie) [inline pkgs/applications/graphics/feh/default.nix:19]: Please use `fetchpath` instead (on https://github.com/derf/feh/commit/2c31f8863b80030e772a529ade519fc2fee4a991.patch)

**Review by ambroisie — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/feh/default.nix (hunk 2 above)
Comment (Artturin) [reply pkgs/applications/graphics/feh/default.nix:19]: Already in the patch I sent

Comment (CharlzKlug) [reply pkgs/applications/graphics/feh/default.nix]: I have added your changes.

**Pull request merged**